pkgs/top-level/all-packages.nix

```
@@ -19691,7 +19691,11 @@ in
 
   calculix = callPackage ../applications/science/math/calculix {};
 
-  calibre = libsForQt5.callPackage ../applications/misc/calibre { };
+  calibre-py2 = libsForQt5.callPackage ../applications/misc/calibre { pythonPackages = python2Packages; };
+
+  calibre-py3 = libsForQt5.callPackage ../applications/misc/calibre { pythonPackages = python3Packages; };
+
+  calibre = calibre-py3;
```

I believe it can make a difference with overlays
```suggestion
  calibre = self.calibre-py3;
```

---

I'll take your word for it. Fixed! Bumped the version while at it. Works here.

---

So grahamcofborg begs to differ:

```
nix-env did not evaluate cleanly:
 ["trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions.", "trace: It seems that you are using a patched Nixpkgs that references the self", "variable in pkgs/top-level/all-packages.nix. This variable was incorrectly", "named, so its usage needs attention. Please use pkgs for packages or super", "for functions."]
```

---

interesting, not what I was expecting, as this works in python packages.